File: house/flutter/lib/models/community.dart
import 'package:hive/hive.dart';

part 'community.g.dart';

@HiveType(typeId: 3)
class Community extends HiveObject {
  @HiveField(0)
  final int id;

  @HiveField(1)
  final String name;

  @HiveField(2)
  final String? officialName;

  @HiveField(3)
  final int lifeCircleId;

  @HiveField(4)
  final int blockId;

  // Basic info
  @HiveField(5)
  final String? address;

  @HiveField(6)
  final double? latitude;

  @HiveField(7)
  final double? longitude;

  @HiveField(8)
  final int? completionYear;

  // Building info
  @HiveField(9)
  final double? landArea;

  @HiveField(10)
  final double? buildingArea;

  @HiveField(11)
  final double? plotRatio;

  @HiveField(12)
  final double? greeningRate;

  @HiveField(13)
  final int? buildingCount;

  @HiveField(14)
  final int? floorCount;

  // Parking
  @HiveField(15)
  final int? parkingTotal;

  @HiveField(16)
  final double? parkingRatio;

  // Traffic
  @HiveField(17)
  final String? nearestMetro;

  @HiveField(18)
  final List<String>? busLines;

  // Units
  @HiveField(19)
  final String? unitTypes;

  @HiveField(20)
  final String? mainUnitType;

  @HiveField(21)
  final String? unitAreaRange;

  @HiveField(22)
  final String? decorationStatus;

  // Pricing
  @HiveField(23)
  final double? listingPrice;

  @HiveField(24)
  final double? dealPrice;

  @HiveField(25)
  final String? priceTrend;

  @HiveField(26)
  final int? daysOnMarket;

  @HiveField(27)
  final double? bargainingSpace;

  @HiveField(28)
  final int? onSaleCount;

  @HiveField(29)
  final int? onRentCount;

  @HiveField(30)
  final String? rentLevel;

  // Education
  @HiveField(31)
  final String? assignedSchool;

  // Environment
  @HiveField(32)
  final String? gardenLandscape;

  @HiveField(33)
  final String? noiseSources;

  @HiveField(34)
  final String? propertyMgmtLevel;

  @HiveField(35)
  final String? security;

  @HiveField(36)
  final bool? pedestrianVehicleSeparation;

  @HiveField(37)
  final String? elevatorBrand;

  @HiveField(38)
  final bool? floodingRisk;

  // Reputation
  @HiveField(39)
  final double? rating;

  @HiveField(40)
  final bool? hasLitigationHistory;

  @HiveField(41)
  final bool? hasNegativeNews;

  Community({
    required this.id,
    required this.name,
    this.officialName,
    required this.lifeCircleId,
    required this.blockId,
    this.address,
    this.latitude,
    this.longitude,
    this.completionYear,
    this.landArea,
    this.buildingArea,
    this.plotRatio,
    this.greeningRate,
    this.buildingCount,
    this.floorCount,
    this.parkingTotal,
    this.parkingRatio,
    this.nearestMetro,
    this.busLines,
    this.unitTypes,
    this.mainUnitType,
    this.unitAreaRange,
    this.decorationStatus,
    this.listingPrice,
    this.dealPrice,
    this.priceTrend,
    this.daysOnMarket,
    this.bargainingSpace,
    this.onSaleCount,
    this.onRentCount,
    this.rentLevel,
    this.assignedSchool,
    this.gardenLandscape,
    this.noiseSources,
    this.propertyMgmtLevel,
    this.security,
    this.pedestrianVehicleSeparation,
    this.elevatorBrand,
    this.floodingRisk,
    this.rating,
    this.hasLitigationHistory,
    this.hasNegativeNews,
  });

  factory Community.fromJson(Map<String, dynamic> json) {
    return Community(
      id: json['id'] as int,
      name: json['name'] as String,
      officialName: json['official_name'] as String?,
      lifeCircleId: json['life_circle_id'] as int,
      blockId: json['block_id'] as int,
      address: json['address'] as String?,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      completionYear: json['completion_year'] as int?,
      landArea: (json['land_area'] as num?)?.toDouble(),
      buildingArea: (json['building_area'] as num?)?.toDouble(),
      plotRatio: (json['plot_ratio'] as num?)?.toDouble(),
      greeningRate: (json['greening_rate'] as num?)?.toDouble(),
      buildingCount: json['building_count'] as int?,
      floorCount: json['floor_count'] as int?,
      parkingTotal: json['parking_total'] as int?,
      parkingRatio: (json['parking_ratio'] as num?)?.toDouble(),
      nearestMetro: json['nearest_metro'] as String?,
      busLines: (json['bus_lines'] as List<dynamic>?)
          ?.map((e) => e as String)
          .toList(),
      unitTypes: json['unit_types'] as String?,
      mainUnitType: json['main_unit_type'] as String?,
      unitAreaRange: json['unit_area_range'] as String?,
      decorationStatus: json['decoration_status'] as String?,
      listingPrice: (json['listing_price'] as num?)?.toDouble(),
      dealPrice: (json['deal_price'] as num?)?.toDouble(),
      priceTrend: json['price_trend'] as String?,
      daysOnMarket: json['days_on_market'] as int?,
      bargainingSpace: (json['bargaining_space'] as num?)?.toDouble(),
      onSaleCount: json['on_sale_count'] as int?,
      onRentCount: json['on_rent_count'] as int?,
      rentLevel: json['rent_level'] as String?,
      assignedSchool: json['assigned_school'] as String?,
      gardenLandscape: json['garden_landscape'] as String?,
      noiseSources: json['noise_sources'] as String?,
      propertyMgmtLevel: json['property_mgmt_level'] as String?,
      security: json['security'] as String?,
      pedestrianVehicleSeparation:
          json['pedestrian_vehicle_separation'] as bool?,
      elevatorBrand: json['elevator_brand'] as String?,
      floodingRisk: json['flooding_risk'] as bool?,
      rating: (json['rating'] as num?)?.toDouble(),
      hasLitigationHistory: json['has_litigation_history'] as bool?,
      hasNegativeNews: json['has_negative_news'] as bool?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'official_name': officialName,
        'life_circle_id': lifeCircleId,
        'block_id': blockId,
        'address': address,
        'latitude': latitude,
        'longitude': longitude,
        'completion_year': completionYear,
        'land_area': landArea,
        'building_area': buildingArea,
        'plot_ratio': plotRatio,
        'greening_rate': greeningRate,
        'building_count': buildingCount,
        'floor_count': floorCount,
        'parking_total': parkingTotal,
        'parking_ratio': parkingRatio,
        'nearest_metro': nearestMetro,
        'bus_lines': busLines,
        'unit_types': unitTypes,
        'main_unit_type': mainUnitType,
        'unit_area_range': unitAreaRange,
        'decoration_status': decorationStatus,
        'listing_price': listingPrice,
        'deal_price': dealPrice,
        'price_trend': priceTrend,
        'days_on_market': daysOnMarket,
        'bargaining_space': bargainingSpace,
        'on_sale_count': onSaleCount,
        'on_rent_count': onRentCount,
        'rent_level': rentLevel,
        'assigned_school': assignedSchool,
        'garden_landscape': gardenLandscape,
        'noise_sources': noiseSources,
        'property_mgmt_level': propertyMgmtLevel,
        'security': security,
        'pedestrian_vehicle_separation': pedestrianVehicleSeparation,
        'elevator_brand': elevatorBrand,
        'flooding_risk': floodingRisk,
        'rating': rating,
        'has_litigation_history': hasLitigationHistory,
        'has_negative_news': hasNegativeNews,
      };
}

File: house/internal/model/community.go
package model

// Community 小区
type Community struct {
	ID            int64   `json:"id" gorm:"primaryKey"`
	Name          string  `json:"name" gorm:"size:100;not null"`        // 小区名称
	OfficialName  string  `json:"official_name" gorm:"size:100"`         // 备案名
	DistrictID    int64   `json:"district_id" gorm:"index"`
	BlockID       int64   `json:"block_id" gorm:"index"`
	LifeCircleID  int64   `json:"life_circle_id" gorm:"index"`
	Address       string  `json:"address" gorm:"size:200"`              // 详细地址
	Coordinates   string  `json:"coordinates" gorm:"size:50"`           // 坐标 lng,lat
	CompletionYear int    `json:"completion_year"`                      // 竣工年份
	BuildingAge   int     `json:"building_age"`                        // 楼龄
	TotalUnits    int     `json:"total_units"`                         // 总户数
	Developer     string  `json:"developer" gorm:"size:100"`           // 开发商
	PropertyComp  string  `json:"property_company" gorm:"size:100"`   // 物业公司
	PropertyFee   float64 `json:"property_fee"`                        // 物业费
	Positioning   string  `json:"positioning" gorm:"size:20"`          // 定位

	// 建筑信息
	LandArea       float64 `json:"land_area"`        // 占地面积 m²
	BuildingArea   float64 `json:"building_area"`    // 建筑面积 m²
	PlotRatio      float64 `json:"plot_ratio"`       // 容积率
	GreeningRate   float64 `json:"greening_rate"`    // 绿化率
	BuildingCount  int     `json:"building_count"`   // 栋数
	FloorCount     int     `json:"floor_count"`      // 层数
	ElevatorRatio  string  `json:"elevator_ratio"`   // 梯户比
	UnitAreaRatio  float64 `json:"unit_area_ratio"`  // 得房率
	OrientationDist string `json:"orientation_distribution" gorm:"type:text"` // 朝向分布

	// 停车位
	ParkingTotal   int     `json:"parking_total"`    // 总车位
	ParkingRatio   string  `json:"parking_ratio"`    // 车户比
	ParkingAbove   int     `json:"parking_above"`    // 地上
	ParkingBelow   int     `json:"parking_below"`    // 地下
	ParkingFeeMonthly float64 `json:"parking_fee_monthly"` // 月租
	ParkingFeeHourly  float64 `json:"parking_fee_hourly"`  // 临停

	// 交通
	NearestMetro    string `json:"nearest_metro" gorm:"type:text"`     // 最近地铁站
	BusLinesCount   int    `json:"bus_lines_count"`                   // 公交线路数

	// 户型
	UnitTypes      string `json:"unit_types" gorm:"type:text"`        // 户型种类
	MainUnitType   string `json:"main_unit_type"`                    // 主力户型
	AreaRange      string `json:"area_range" gorm:"type:text"`       // 面积段
	DecorationStatus string `json:"decoration_status" gorm:"size:10"` // 装修状况

	// 价格行情
	ListingPrice      float64 `json:"listing_price"`        // 挂牌价
	RecentDealPrice   float64 `json:"recent_deal_price"`   // 成交价
	PriceTrend        string  `json:"price_trend" gorm:"type:text"` // 历史走势
	AvgDaysOnMarket   int     `json:"avg_days_on_market"`  // 平均成交周期
	BargainingSpace   float64 `json:"bargaining_space"`    // 议价空间
	OnSaleCount       int     `json:"on_sale_count"`       // 在售
	OnRentCount       int     `json:"on_rent_count"`       // 在租
	RentLevel         string  `json:"rent_level" gorm:"type:text"` // 租金

	// 教育配套
	AssignedSchool string `json:"assigned_school" gorm:"type:text"` // 对口学校

	// 环境与安全
	GardenLandscape        string `json:"garden_landscape"`       // 园林景观
	NoiseSources          string `json:"noise_sources" gorm:"type:text"` // 噪音源
	PropertyMgmtLevel     string `json:"property_management_level" gorm:"size:10"`
	SecuritySystem       string `json:"security_system"`       // 安防系统
	PedestrianVehicleSep  bool   `json:"pedestrian_vehicle_separation"` // 人车分流
	ElevatorBrand        string `json:"elevator_brand"`        // 电梯品牌
	FloodingRisk         bool   `json:"flooding_risk"`         // 内涝风险

	// 不利因素
	AdverseFactors500m  string `json:"adverse_factors_500m" gorm:"type:text"`  // 500米内
	AdverseFactors1000m string `json:"adverse_factors_1000m" gorm:"type:text"` // 1000米内

	// 口碑
	Rating           float64 `json:"rating"`             // 评分
	LitigationHistory string `json:"litigation_history" gorm:"type:text"` // 维权记录
	NegativeNews     string `json:"negative_news" gorm:"type:text"`     // 负面新闻

	// 关联
	District   *District  `json:"district,omitempty" gorm:"foreignKey:DistrictID"`
	Block      *Block     `json:"block,omitempty" gorm:"foreignKey:BlockID"`
	LifeCircle *LifeCircle `json:"life_circle,omitempty" gorm:"foreignKey:LifeCircleID"`
}

// TableName 返回表名
func (Community) TableName() string {
	return "communities"
}
